src/audio/load.c

```diff
@@ -851,105 +852,118 @@ void load_sequence_internal(u32 player, u32 seqId, s32 loadAsync) {
+        );
 #endif
 
-    eu_stubbed_printf_1("AudioHeap is %x\n", gAudioHeapSize);
+        eu_stubbed_printf_1("AudioHeap is %x\n", gAudioHeapSize);
 
-    for (i = 0; i < NUMAIBUFFERS; i++) {
-        gAiBufferLengths[i] = 0xa0;
-    }
+        for (i = 0; i < NUMAIBUFFERS; i++) {
+            gAiBufferLengths[i] = 0xa0;
+        }
 
-    gAudioFrameCount = 0;
-    gAudioTaskIndex = 0;
-    gCurrAiBufferIndex = 0;
-    gSoundMode = 0;
-    gAudioTask = NULL;
-    gAudioTasks[0].task.t.data_size = 0;
-    gAudioTasks[1].task.t.data_size = 0;
-    osCreateMesgQueue(&gAudioDmaMesgQueue, &gAudioDmaMesg, 1);
-    osCreateMesgQueue(&gCurrAudioFrameDmaQueue, gCurrAudioFrameDmaMesgBufs,
-                      ARRAY_COUNT(gCurrAudioFrameDmaMesgBufs));
-    gCurrAudioFrameDmaCount = 0;
-    gSampleDmaNumListItems = 0;
-
-    sound_init_main_pools(gAudioInitPoolSize);
-
-    bzero(&gAiBuffers, sizeof(gAiBuffers));
-    for (i = 0; i < NUMAIBUFFERS; i++) {
-        gAiBuffers[i] = soundAlloc(&gAudioInitPool, AIBUFFER_LEN);
-
-        /*for (j = 0; j < (s32) (AIBUFFER_LEN / sizeof(s16)); j++) {
-            gAiBuffers[i][j] = 0;
-        }*/
-    }
+        gAudioFrameCount = 0;
+        gAudioTaskIndex = 0;
+        gCurrAiBufferIndex = 0;
+        gSoundMode = 0;
+        gAudioTask = NULL;
+        gAudioTasks[0].task.t.data_size = 0;
+        gAudioTasks[1].task.t.data_size = 0;
+        osCreateMesgQueue(&gAudioDmaMesgQueue, &gAudioDmaMesg, 1);
+        osCreateMesgQueue(&gCurrAudioFrameDmaQueue, gCurrAudioFrameDmaMesgBufs,
+                          ARRAY_COUNT(gCurrAudioFrameDmaMesgBufs));
+        gCurrAudioFrameDmaCount = 0;
+        gSampleDmaNumListItems = 0;
+
+        sound_init_main_pools(gAudioInitPoolSize);
+
+        bzero(&gAiBuffers, sizeof(gAiBuffers));
+        for (i = 0; i < NUMAIBUFFERS; i++) {
+            gAiBuffers[i] = soundAlloc(&gAudioInitPool, AIBUFFER_LEN);
+
+            /*for (j = 0; j < (s32) (AIBUFFER_LEN / sizeof(s16)); j++) {
+                gAiBuffers[i][j] = 0;
+            }*/
+        }
 
 #if defined(VERSION_EU)
-    gAudioResetPresetIdToLoad = 0;
-    gAudioResetStatus = 1;
-    audio_shut_down_and_reset_step();
+        gAudioResetPresetIdToLoad = 0;
+        gAudioResetStatus = 1;
+        audio_shut_down_and_reset_step();
 #else
-    audio_reset_session(&gAudioSessionPresets[0], 0);
+        audio_reset_session(&gAudioSessionPresets[0], 0);
 #endif
 
-    // Not sure about these prints
-    eu_stubbed_printf_1("Heap reset.Synth Change %x \n", 0);
-    eu_stubbed_printf_3("Heap %x %x %x\n", 0, 0, 0);
-    eu_stubbed_printf_0("Main Heap Initialize.\n");
+        // Not sure about these prints
+        eu_stubbed_printf_1("Heap reset.Synth Change %x \n", 0);
+        eu_stubbed_printf_3("Heap %x %x %x\n", 0, 0, 0);
+        eu_stubbed_printf_0("Main Heap Initialize.\n");
+    }
+
+    if (initType == AUD_REINIT) {
+        sound_init_main_pools(gAudioInitPoolSize);
```

Comment:
This seems bad; since `sAudioIsInitialized` doesn't appear to be changed back to FALSE anywhere, and since there's no following call to `audio_reset_session`, I have to assume the audio isn't exactly being reinitialized here (and thus use of this function is a no-no). The only thing that's effectively happening here I'm pretty sure is clearing of `gAiBuffers` as a side effect of soundAlloc, so you should only need to uncomment that code block and delete everything else in this function (or even better, just bzero the whole thing).

Note I'm not referring to `bzero(&gAiBuffers, sizeof(gAiBuffers));`, this appears to be a Fazana moment and pretty much has no reason to exist in the unchanged function as far as I can tell. (In fact, please just delete line 933 where it happens earlier.)

Reply:
Honestly it's also probably for the best if what's currently outlined here happens in a different function. `audio_init` should just be left alone if it's not truly being reinitialized.

Reply:
UPDATE: I'm a moron, I didn't expand the function and forgot the audio init allocations took place in the same function. That said, still don't use that function. Just manually update the `gAudioInitPool`'s cur pointer back to start.